text/0000-official-utility-buildpacks.md

```diff
@@ -0,0 +1,75 @@
+# Meta
+[meta]: #meta
+- Name: Officially supported utility buildpacks
+- Start Date: 2021-06-30
+- Author(s): [@jkutner](https://github.com/jkutner)
+- RFC Pull Request: (leave blank)
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+- Supersedes: N/A
+
+# Summary
+[summary]: #summary
+
+This is a proposal to adopt utility buildpacks as officially supported components of the Cloud Native Buildpacks project.
+
+# Definitions
+[definitions]: #definitions
+
+- **utility buildpack** - a buildpack that provides a generic capability that is useful across many vendors, platforms, and apps
+- **officially supported** - supported and maintained by the the Cloud Native Buildpacks project and its members
+
+# Motivation
+[motivation]: #motivation
+
+Many capabilities of lifecycle and pack do not require a native implementation within these components. These capabilities can (and have been) implemented at the user layer as buildpacks (ex. [inline-buildpack](https://registry.buildpacks.io/buildpacks/hone/inline)). In [PR #168](https://github.com/buildpacks/rfcs/pull/168) we assert that supporting these capabilities natively adds complexity to the implementation of lifecycle, which makes it difficult to innovate and improve.
+
+However, defering to the community and vendors to implement these kinds of utility buildpacks has not necessarily had good results. For example, there are three implementations of the [Procfile buildpack](https://registry.buildpacks.io/buildpacks/paketo-buildpacks/procfile). None of these implementations add significant (if any) advantages over the other. Yet buildpack users are force to decide which one to select. This is an unneccessary burden for a capability that could be provided by default.
+
+We seek to reduce complexity in the lifecycle and provide well maintained utility buildpacks for users by supporting them as official components of the Cloud Native Buildpacks project. This also aligns with our "everything's a buildpack" mantra.
+
+# What it is
+[what-it-is]: #what-it-is
+
+Reserve the `cnb` namespace for officially supported buildpacks. Then, on a case by case basis, we adopt utility buildpacks under this
+
+Examples of potential utility buildpacks include:
+
+- `cnb/profile` - Evaluating `.profile` scripts in the app repo during launch
+- `cnb/inline` - Executing [Inline buildpacks](https://github.com/buildpacks/rfcs/blob/main/text/0048-inline-buildpack.md)
```

Comment:
I'm a bit confused how this would work - it seems that the inline script would need to be copied to the proper directory under `/cnb/buildpacks` ?? That doesn't feel like a buildpack concern...

Reply:
I'm not sure I follow. You can see an implementation of this works in https://github.com/hone/inline-buildpack

Reply:
Ohh I see (thank you for linking to the source!). The "inline wrapper" buildpack when run looks in project.toml and runs the script there. I was confusing it with pack's implementation, where the script is copied (I think?).